src/base/bittorrent/session.cpp

```diff
@@ -724,6 +770,45 @@ void Session::setSubcategoriesEnabled(bool value)
     emit subcategoriesSupportChanged();
 }
 
+QSet<QString> Session::tags() const
+{
+    return m_tags;
+}
+
+bool Session::isValidTag(const QString &tag)
+{
+    return !tag.trimmed().isEmpty() && !tag.contains(",");
```

Comment:
String-to-char: Done.

Curious, why do you prefer wrapping the return statement?

About strict tags, I would really prefer not restricting characters too much. When using similar systems in other torrent clients, I have often found it useful to be able to include non-alphanum symbols. Just as one example, there is a band named "!!!". I can see the motivation for restricting the maximum length because super long tags could cause issues with the UI. Do you have a specific value in mind? I don't see any limit on category lengths, but maybe I am missing it.